Qonversion/Qonversion/NetworkLayer/RequestProcessor/RequestProcessor.swift

```diff
@@ -11,23 +11,35 @@ class RequestProcessor: RequestProcessorInterface {
     let headersBuilder: HeadersBuilderInterface
     let errorHandler: NetworkErrorHandler
     let decoder: ResponseDecoderInterface
+    let retriableRequestsList: [Request]
+    var criticalError: QonversionError?
     
-    init(baseURL: String, networkProvider: NetworkProvider, headersBuilder: HeadersBuilderInterface, errorHandler: NetworkErrorHandler, decoder: ResponseDecoderInterface) {
+    init(baseURL: String, networkProvider: NetworkProvider, headersBuilder: HeadersBuilderInterface, errorHandler: NetworkErrorHandler, decoder: ResponseDecoderInterface, retriableRequestsList: [Request]) {
         self.baseURL = baseURL
         self.networkProvider = networkProvider
         self.headersBuilder = headersBuilder
         self.errorHandler = errorHandler
         self.decoder = decoder
+        self.retriableRequestsList = retriableRequestsList
     }
     
     func process<T>(request: Request, responseType: T.Type) async throws -> T? where T : Decodable {
+        guard criticalError == nil else { throw criticalError! }
```

Comment:
Maybe so, as we did in the `DeviceInfoCollector`?
```suggestion
    if let savedCriticalError: QonversionError = criticalError {
      return savedCriticalError
    }
```